alacritty/src/event.rs

```diff
@@ -414,6 +428,11 @@ impl<N: Notify> Processor<N> {
                 }
             }
 
+            // Process font change events
+            if !glyph_cache_refresh_pending.is_empty() {
+                self.display.handle_glyph_cache_refresh(&self.config);
+            }
```

Comment:
Okay, bear with me here but I think we can do better than having another method that is weirdly strung through our event loop here. The resize should be sufficient.

If you take a look at the font resize code, you'll see that it already reloads the font completely. It also refills the cache with the new glyphs. If I'm reading this correctly, that should mean that our code for changing size and changing font should be identical.

The only thing we should need to change is to pass a `Font` to resize instead of a `Size`. So this would allow us to skip this whole `GlyphCacheRefresh` and solely use the `font_size` to change both font size and font family.

I'm a bit conflicted about naming it `font_size` if it does more than just resizing and I think even naming it `Resize` when it does more than resizing is a bit suboptimal. But our best approach should be to rename the font resize from `font_size` to just `font` and adding a comment that this is a font resize or family change.

If you have any questions about this, please let me know, but hopefully this should eliminate the need for separately handling font size and family changes.

Reply:
I think the approach you describe sounds totally reasonable. I did miss that `update_font_cache` is essentially reloading the font anyway. Are you okay with me renaming some things to make it all semantically correct? ie `Resize`, `resize_pending`, etc

Reply:
What would you rename `Resize`, `resize_pending` and `handle_resize` to? I'm fine with finding better names, however I'm not sure if that's so simple.

Reply:
The common theme I see is that the thing being updated is the configuration of the `Display`, so I was thinking:

* `Resize` -> `ReconfigureDisplay`
* `resize_pending` -> `display_reconfigure_pending`
* `handle_resize` -> `handle_reconfigure`

I also considered something along the lines of `UpdateDisplay`, but that seems to me like it could be misunderstood to refer to a per-loop-iteration update of the display's contents.

`UpdateDisplayConfig` might be a good fit but feels it would get a bit long-winded   since we would end up with `update_display_config_pending`.

Reply:
We're not really changing any display configuration though. The `Display` is not an actual display, it's a struct responsible for displaying content.

Also since we're talking about an enum, we should name things appropriately. `UpdateDisplay` might be a trait name, but an enum should be named `DisplayUpdate`. The enum isn't actually doing anything, it's storing information to do something, so we can't name it like it does something itself.

We also don't only update the display or renderer in this method. The resize is also responsible for updating the terminal and PTY sizes, so naming it something specifically related to the renderer or display is very misleading. It's not stateless outside of `display.rs`.

Reply:
Ah shoot, I forgot to submit this comment last week...

> Also since we're talking about an enum, we should name things appropriately. UpdateDisplay might be a trait name, but an enum should be named DisplayUpdate

Good point, I renamed it `DisplayUpdate` in my latest set of updates.

> We also don't only update the display or renderer in this method. The resize is also responsible for updating the terminal and PTY sizes, so naming it something specifically related to the renderer or display is very misleading.

The only place I saw the `Resize` (now `DisplayUpdate`) enum being used was to pass information to the `Display` struct's `handle_resize` method. Could you point out to me where else it is being used to pass update information along?

Reply:
`Resize` isn't used in other places, but `handle_resize` does other things too.